internal/operators/ocs/validations.go

```diff
@@ -128,40 +123,56 @@ func (o *ocsValidator) ValidateRequirements(cluster *models.Cluster) (api.Valida
+		return api.Failure, errors.New("Error occurred while calculating Node requirements ")
+	} else if inventoryMissing {
+		return api.Pending, errors.New("Missing Inventory in some of the hosts ")
+	}
+	return "", nil
+}
+
+func (o *operator) countNodeResources(r resources) (bool, error) {
 	var inventory models.Inventory
 	// if inventory is empty
-	if host.Inventory == "" {
-		o.log.Info("Empty Inventory of host with hostID ", *host.ID)
+	if r.host.Inventory == "" {
+		o.log.Info("Empty Inventory of host with hostID ", *r.host.ID)
 		return true, nil // to indicate that inventory is empty and the ValidationStatus must be Pending
-	} else if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
-		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+	} else if err := json.Unmarshal([]byte(r.host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", r.host.ID)
 		return false, err
 	}
 
 	disks := getValidDiskCount(inventory.Disks)
 
 	if disks > 1 { // OCS must use the non-boot disks
-		requiredDiskCPU := int64(disks-1) * o.OCSRequiredDiskCPUCount
-		requiredDiskRAM := int64(disks-1) * o.OCSRequiredDiskRAMGB
-
-		if inventory.CPU.Count < requiredDiskCPU || inventory.Memory.UsableBytes < gbToBytes(requiredDiskRAM) {
-			status := fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " RAMGB")
-			*insufficientHosts = append(*insufficientHosts, status)
-			o.log.Info(status)
-		} else {
-			*cpuCount += (inventory.CPU.Count - requiredDiskCPU)                     // cpus excluding the cpus required for disks
-			*totalRAM += (inventory.Memory.UsableBytes - gbToBytes(requiredDiskRAM)) // ram excluding the ram required for disks
-			*diskCount += (int64)(disks - 1)                                         // not counting the boot disk
-			*hostsWithDisks++
-		}
+		o.countResourcesExcludingDiskResources(&inventory, r, int64(disks))
 	} else {
-		*cpuCount += inventory.CPU.Count
-		*totalRAM += inventory.Memory.UsableBytes
+		*r.cpuCount += inventory.CPU.Count
+		*r.totalRAM += inventory.Memory.UsableBytes
 	}
 	return false, nil
 }
 
+func (o *operator) countResourcesExcludingDiskResources(inventory *models.Inventory, r resources, disks int64) {
+	// OCS requires 2 cpus, 5GB ram per disk
+	requiredDiskCPU := (disks - 1) * o.config.OCSRequiredDiskCPUCount
+	requiredDiskRAM := (disks - 1) * o.config.OCSRequiredDiskRAMGB
+
+	if inventory.CPU.Count < requiredDiskCPU || inventory.Memory.UsableBytes < gbToBytes(requiredDiskRAM) {
+		status := fmt.Sprint("Insufficient resources on host with host ID ", *r.host.ID, " to deploy OCS. The hosts has ", disks, " disks that require ", requiredDiskCPU, " CPUs, ", requiredDiskRAM, " RAMGB")
+		*r.insufficientHosts = append(*r.insufficientHosts, status)
+		o.log.Info(status)
```

Comment:
Unrelated to this patch but shouldn't this error out here?

Reply:
We wanted to show all the hosts that did not have sufficient disk resources to the user